tests/testlibrary.c

```diff
@@ -327,6 +354,125 @@ test_list_refs (void)
   g_assert_cmpint (refs->len, ==, 0);
 }
 
+static void
+create_multi_collection_id_repo (const char *repo_dir)
+{
+  int status;
+  g_autoptr(GError) error = NULL;
+  GSpawnFlags flags = G_SPAWN_DEFAULT;
+  g_autofree char *arg0 = NULL;
+  g_autofree char *argv_str = NULL;
+
+  /* Create a repository in which each app has a different collection-id */
+  arg0 = g_test_build_filename (G_TEST_DIST, "make-multi-collection-id-repo.sh", NULL);
+  const char *argv[] = { arg0, repo_dir, NULL };
+  argv_str = g_strjoinv (" ", (char **) argv);
+
+  if (g_test_verbose ())
+    g_print ("running %s\n", argv_str);
+  else
+    flags |= G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL;
+
+  g_test_message ("Spawning %s", argv_str);
+  g_spawn_sync (NULL, (char **) argv, NULL, flags, NULL, NULL, NULL, NULL, &status, &error);
+  g_assert_no_error (error);
+  g_assert_cmpint (status, ==, 0);
+}
+
+#ifdef FLATPAK_ENABLE_P2P
+static void
+test_list_refs_in_remotes (void)
+{
+  int status;
+  const char *repo_name = "multi-refs-repo";
+  GSpawnFlags flags = G_SPAWN_SEARCH_PATH;
+  g_autofree char *argv_str = NULL;
+  g_autofree char *repo_url = NULL;
+  g_autoptr(GPtrArray) refs1 = NULL;
+  g_autoptr(GPtrArray) refs2 = NULL;
+  g_autoptr(FlatpakInstallation) inst = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(FlatpakRemote) remote = NULL;
+  g_autofree char *repo_dir = g_build_filename (testdir, repo_name, NULL);
+  g_autofree char *repo_uri = NULL;
+  g_autoptr(GHashTable) collection_ids = g_hash_table_new_full (g_str_hash,
+                                                                g_str_equal,
+<<<<<<< HEAD
```

Comment:
Seems to be some rebase noise here. Did this get compile-tested?

Reply:
O_O I have compiled things a few times this very morning. Maybe I pushed the wrong branch. Let me check.

Reply:
Yup. I have about 5 branches because of the rebase I had to do in December... Sorry for the mess.